method/group/init.js

```diff
@@ -37,18 +32,24 @@ module.exports = function GroupCreate (ssb, _, state) {
     //   - need to check if it's safe to make a sharedDM with oneself
     // would also require adding groupKey to this message
 
-    const envelope = box(plain, state.feedId, state.previous, msgKey, recipientKeys)
-    const ciphertext = envelope.toString('base64') + '.box2'
-
-    ssb.publish(ciphertext, (err, groupInitMsg) => {
+    ssb.getFeedState(ssb.id, (err, previousFeedState) => {
       if (err) return cb(err)
 
-      const data = {
-        groupId: groupId({ groupInitMsg, msgKey }),
-        groupKey: groupKey.toBuffer(),
-        groupInitMsg
-      }
-      cb(null, data)
+      const previousMessageId = new MsgId(previousFeedState.id).toTFK()
+
+      const envelope = box(plain, state.feedId, previousMessageId, msgKey, recipientKeys)
+      const ciphertext = envelope.toString('base64') + '.box2'
+
+      ssb.publish(ciphertext, (err, groupInitMsg) => {
+        if (err) return cb(err)
+
+        const data = {
+          groupId: groupId({ groupInitMsg, msgKey }),
+          groupKey: groupKey.toBuffer(),
+          groupInitMsg
+        }
```

Comment:
TODO - we discussed a final (paranoid) step where we check auto-unboxing works ... (as it's possible for someone to have queued another message before us which we are encrypting based on an (_now outdated_) `previous`.

I think the place for that is in the `index.js` after we register the group?

Reply:
DONE